guide/src/class/object.md

```diff
@@ -217,8 +217,8 @@ impl Number {
 
 # fn main() -> PyResult<()> {
 #     Python::with_gil(|py| {
-#         let x = PyCell::new(py, Number(4))?;
-#         let y = PyCell::new(py, Number(4))?;
+#         let x = &Bound::new(py, Number(4))?.into_any();
```

Comment:
These `.as_any()` and `.into_any()` calls are probably a consequence of the same thing as https://github.com/PyO3/pyo3/pull/3867#discussion_r1493846842

I suggest we merge here anyway and then can tidy up a bunch of `.as_any()` and `.into_any()` calls after figuring out a solution.

Reply:
Yes, I've just added some thoughts to that thread.